### PR TITLE
Add hash functions

### PR DESCRIPTION
**Pull request by havatv**
Adds the list of hash functions introduced in the commit referenced in #4331. No explanations or examples. Perhaps that can be added later?

Goal:

Ticket(s): #

- [ ] Backport to LTR documentation is required

### CODE REVIEW

**Comment by Gustry** [comment]
This is for 3.12, so I guess this PR should be merged only after the 3.10 branch is created. No?

**Comment by havatv** [comment]
Thanks, @Gustry! I added the 3.12 label.

File: source/docs/user_manual/working_with_vector/expression.rst [hunk 1]
```diff
@@ -759,6 +759,43 @@ This group contains functions that operate on geometry objects (e.g., length, ar
     CASE WHEN $area > 10 000 THEN 'Larger' ELSE 'Smaller' END
 
 
+Hash Functions
+----------------
+
+This group contains hash functions.
+All these functions return the hash (a string hex value) of the string argument.
+
+.. csv-table::
```
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/expression.rst]: Should be better to make a simple list of supported "formats" than a table with a single filled column and an empty one

Comment (havatv) [reply source/docs/user_manual/working_with_vector/expression.rst]: In case someone thinks a description of the hash functions would be a useful thing.

**Comment by havatv** [comment]
Descriptions have now been added for all the functions.
I will merge in a couple of days if there are no objections.

**Pull request merged**